mcs/class/System.Data/Test/ProviderTests/System.Data.SqlClient/SqlConnectionTest.cs

```diff
@@ -287,12 +287,22 @@ public void Close ()
 		[Test]
 		public void ChangeDatabase ()
 		{
-			if (ConnectionManager.Instance.Sql.IsAzure)
-				Assert.Ignore("SQL Azure doesn't support 'ChangeDatabase'");
+			conn = new SqlConnection(connectionString);
+			conn.Open();
 
-			conn = new SqlConnection (connectionString);
-			conn.Open ();
-			conn.ChangeDatabase ("master");
+			bool isAzure = ConnectionManager.Instance.Sql.IsAzure;
+
+			try
+			{
+				conn.ChangeDatabase("master");
+			}
+			catch (SqlException exc) when (isAzure)
+			{
+				Assert.AreEqual(40508, exc.Number); //USE statement is not supported to switch between databases (Azure).
+				return;
+			}
+
+			Assert.IsFalse(isAzure);
```

Comment:
this can be simplified to:

```csharp
if (ConnectionManager.Instance.Sql.IsAzure) {
   var exc = Assert.Throws<SqlException> (() => conn.ChangeDatabase ("master"));
   Assert.Equals (40508, exc.Number)
}
else {
   conn.ChangeDatabase ("master")
   Assert.AreEqual ("master", conn.Database);
}
```

(also coding convention: space before opening brace)